src/averageAges.js

```diff
@@ -52,7 +73,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let sum = 0;
+  let counter = 0;
+  const women = people.filter(person => person.sex === 'f');
+
+  women.map(woman => people
+    .map(child => {
+      if (child.mother === woman.name) {
+        if (arguments.length === 1 || child.sex === 'm') {
+          sum += child.born - woman.born;
+          counter++;
+        }
+      }
+    }));
```

Comment:
map should return something without side effects

Reply:
Here I'm using map as a substitute for a 'for loop', I don't even store the result of mapping in the var, so inputing any return would go to nowhere anyway. In normal project, I think I would use a for loop in this case instead of mapping as I actually do not need an array. Maybe I'm wrong)